src/js/libgeo/GeoRender.js

```diff
@@ -99,6 +104,27 @@ function drawgeoline(el) {
         if (zz.value.real >= 0) { // finite segment
             evaluator.draw$2(
                 [el.startpos, el.endpos], modifs);
+            if (el.labeled && !el.tmp) {
+                var lbl = el.printname || el.name || "S";
+                var l = List.cross(el.startpos, el.endpos);
+                var npos = {
+                    'x': l.value[0].value.real,
```

Comment:
Mhmmm actually I don't get why this works at all? And shouldn't there be a check for complex values?

Reply:
Forget about the comment concerning the complex numbers, the check is done via ``!List._helper.isAlmostReal(el.homog)``